PD-38 - Add assets docs (#40)
ultravioletrs/prism-docs: docs/assets.md

# Computation Assets Overview

> Computation assets on Prism provide a convenient way to manage essential computational resources—datasets and algorithms. These assets facilitate basic operations such as creation, reading, updating, and deletion (CRUD). They can also be associated with or unlinked from a computation.

Reviewer: ```suggestion
Computation assets on Prism provide a convenient way to manage essential computational resources including datasets and algorithms.  They can also be associated with or unlinked from a computation.
```

> An important distinction is that computation assets contain file samples of the algorithm or dataset, providing users with a guide on what the actual files will resemble. These samples are **not** the actual files sent to the computation manifest when uploading via the CLI. For this reason, the actual file’s checksum is required during asset creation.

Reviewer: shared with collaborators in a computation

A single asset can be used in multiple computations (one-to-many relationship), and a computation can be linked to several assets.

## Asset Usage

### 1. **Creating a New Asset**

To create a new asset:

1. Navigate to the **Assets** page.
2. Click the **New Asset** button to open the asset creation modal.

   ![Create asset](img/newAsset.png)

The only mandatory fields during asset creation are:

> - **File Hash**: The checksum of the actual file (not the sample).

Reviewer: checksum and link to cli docs for this specific example

- **Asset Description**: A brief summary of what the asset represents.

> The **File Name** field is optional. If provided, it will be used as the file name when running the computation (not the name of the sample file).

Reviewer: it will be verified when asset is uploaded on computation run

2. Click the **Associate** button on the asset.
3. In the search bar, search for the computation to associate the asset with.

   ![Associate Computation](img/asset_association.png)

4. Once the correct computation is found, select it and click **Associate**.

   ![Associated Computations](img/asset_computations.png)

Multiple computations can be linked to a single asset. All associated computations will be listed as pills in the computation(s) section.

To **unlink** an asset from a computation, click the **Unlink** button next to the computation you wish to de-associate from the asset.

### 3. **Viewing and Downloading Asset Samples**

If an asset sample has been uploaded, users can either view it directly on Prism or download it for inspection later. This can be done by clicking the asset name or the **View** button in the assets list.

![View Asset](img/asset_view.png)

### 4. **Running a Computation**

Once all assets have been successfully associated with a computation, they will be listed within the computation’s structure. At this point, the user is required to upload their **public keys** to proceed.

![Computation Page](img/asset_computation.png)

> After the public keys and other necessary requirements have been provided, the computation can be executed. During execution, the computation service will retrieve the associated assets from the database and add them to the computation manifest, which is then sent to the manager.

Reviewer: backend not manager

The only mandatory fields during asset creation are:

> - **Checksum**: The checksum of the actual file (not the sample), get more information on checksum [here](https://docs.cocos.ultraviolet.rs/cli/#file-hash).

Reviewer: please rename to file-checksum

![View Asset](img/asset_view.png)

### 4. **Viewing a computation's assets**

> To view assets associated to a computation, click the **Asset Management** which opens the assets page containing the computation's assets.

Reviewer: asset management button
